docs/dev/install.rst

```diff
@@ -4,79 +4,114 @@ Development Installation
+
+* Is Unix-like system (Linux, BSD, Mac OSX) which **supports Docker**. Windows systems should have WSL+Docker or Docker Desktop.
+* Has ``>=10 GB`` of **free disk space** on the drive where Docker's cache and volumes are stored. If you want to experiment with customizing Docker containers, you'll likely need more.
+* Can spare ``2 GB`` of **system memory** for running Read the Docs, this typically means that a development laptop should have ``>=8 GB`` of memory in total.
 
 .. note::
 
-   We do not recommend to follow this guide to deploy an instance of Read the Docs for production usage.
-   Take into account that this setup is only useful for developing purposes.
+   Take into account that this setup is intended for development purposes. We do not recommend to follow this guide to deploy an instance of Read the Docs for production.
+
+
+Installing Docker
+-----------------
+
+You need `Docker <https://www.docker.com/>`_ to run a development setup of Read the Docs. The easiest place to start is `Docker's installation guide <https://docs.docker.com/install/>`_, however there might be other more specific ways to get Docker for your exact system.
+
+If you run Windows, you can run Docker on `Windows Subsystem for Linux (WSL) <https://docs.microsoft.com/en-us/windows/wsl/tutorials/wsl-containers>`_ or `Docker Desktop for Windows <https://docs.docker.com/desktop/windows/install/>`_.
+
+   .. tip::
+
+      We always support Docker CE for the latest Ubuntu LTS. On Ubuntu 22.04, we are therefore supporting Docker 20.x, the version of the ``docker.io`` package.
+
+      The following will install Docker Community Edition maintained by a Ubuntu/Canonical team:
+
+      .. prompt:: bash
+
+         # Install the docker.io package
+         sudo apt install docker.io
+         # Add the current user to the Docker group. After this, you should log out and back in for the group membership to be effective.
+         sudo adduser $(whoami) docker
 
 
 Set up your environment
 -----------------------
 
-#. install `Docker <https://www.docker.com/>`_ following `their installation guide <https://docs.docker.com/install/>`_.
-
-#. clone the ``readthedocs.org`` repository:
+#. Clone the ``readthedocs.org`` repository:
 
    .. prompt:: bash
 
       git clone --recurse-submodules https://github.com/readthedocs/readthedocs.org/
 
-#. install the requirements from ``common`` submodule:
+#. Install the requirements from ``common`` submodule:
 
    .. prompt:: bash
 
       pip install -r common/dockerfiles/requirements.txt
 
-#. build the Docker image for the servers:
+#. Build the Docker image for the servers:
 
    .. warning::
 
       This command could take a while to finish since it will download several Docker images.
 
    .. prompt:: bash
 
-      inv docker.build
+      COMPOSE_PROJECT_NAME=community inv docker.build
```

Comment:
We should fix the environment variable to not be required on these commands instead of telling users to prepend them every time they need to run `inv`.

I'd guess there is something different in your environment that this is not working. We've seen this problem in the past when people were using a different version of Docker Compose that the one installed in the previous step. Currently, 1.29.2. Are you using this specific version?

Reply:
Also, the version of docker compose comes from `common/dockerfiles/requirements.txt`

Reply:
This is actually going to become important.

docker-compose on PyPi was stopped in V1 and rewritten in Go for V2.

https://pypi.org/project/docker-compose/
vs.
https://github.com/docker/compose/

> Docker Compose V2 is a major version bump release of Docker Compose. It has been completely rewritten from scratch in Golang (V1 was in Python). The installation instructions for Compose V2 differ from V1. V2 is not a standalone binary anymore, and installation scripts will have to be adjusted. Some commands are different.

Reply:
Issue with environment variable fixed here: https://github.com/readthedocs/common/pull/145

Reply:
Once https://github.com/readthedocs/common/pull/145 is merged, we can remove this part of the changeset.

Reply:
@benjaoming 

> docker-compose on PyPi was stopped in V1 and rewritten in Go for V2.

This is unfortunate 😞 . I think it will complicate a little more our initial development setup. We should probably delegate this step to the official documentation as well. (e.g. in Arch, this is `pacman -S docker-compose` and it seems that for Ubuntu it comes when installing the Docker Engine server?)

Reply:
@stsewd 

> Also, the version of docker compose comes from common/dockerfiles/requirements.txt

It seems we upgraded the version from that file without double checking the `.env` file was working properly. The latest version of docker-compose that read that file properly is `1.27.4`. So, we should downgrade that requirement instead of adding `COMPOSE_PROJECT_NAME` everywhere.

> Docker Compose versions earlier than 1.28, load the .env file from the current working directory, where the command is executed, or from the project directory if this is explicitly set with the --project-directory option

(from https://docs.docker.com/compose/env-file/)

---

@benjaoming 

We need to change the order of the `-f` passed to use `-f docker-compose.override.yml` as the first one. That way, it will load the `.env` file automatically. See

> The project directory is specified by the order of precedence:
>
>     --project-directory flag
>     Folder of the first --file flag
>     Current directory

I've tested this with Docker Compose v1.29.2 _and_ Docker Compose v2 and it keeps working fine with `.env` file in the repository. In the future, when we upgrade to Docker Compose v2 we won't need touch/change anything else.

Reply:
If we change the order it won't override the values anymore

> You can supply multiple -f configuration files. When you supply multiple files, Compose combines them into a single configuration. Compose builds the configuration in the order you supply the files. Subsequent files override and add to their predecessors.

https://docs.docker.com/compose/reference/#specifying-multiple-compose-files

Reply:
Adding `--project-directory=.` seems to work for me